notebooks/ch15/bias_cv.md

```diff
@@ -0,0 +1,498 @@
+
+ice = pd.concat([temp, ice])
+```
+
+
+```python
+ice
+```
+
+
+
+
+<div>
+<style scoped>
+    .dataframe tbody tr th:only-of-type {
+        vertical-align: middle;
+    }
+
+    .dataframe tbody tr th {
+        vertical-align: top;
+    }
+
+    .dataframe thead th {
+        text-align: right;
+    }
+</style>
+<table border="1" class="dataframe">
+  <thead>
+    <tr style="text-align: right;">
+      <th></th>
+      <th>sweetness</th>
+      <th>overall</th>
+    </tr>
+  </thead>
+  <tbody>
+    <tr>
+      <th>0</th>
+      <td>3.60</td>
+      <td>3.09</td>
+    </tr>
+    <tr>
+      <th>1</th>
+      <td>3.50</td>
+      <td>3.17</td>
+    </tr>
+    <tr>
+      <th>2</th>
+      <td>3.69</td>
+      <td>3.46</td>
+    </tr>
+    <tr>
+      <th>...</th>
+      <td>...</td>
+      <td>...</td>
+    </tr>
+    <tr>
+      <th>6</th>
+      <td>11.00</td>
+      <td>5.90</td>
+    </tr>
+    <tr>
+      <th>7</th>
+      <td>11.70</td>
+      <td>5.50</td>
+    </tr>
+    <tr>
+      <th>8</th>
+      <td>11.90</td>
+      <td>5.40</td>
+    </tr>
+  </tbody>
+</table>
+<p>309 rows × 2 columns</p>
+</div>
+
+
+
+
+```python
+# HIDDEN
+plt.scatter(ice['sweetness'], ice['overall'], s=10)
+plt.title('Ice Cream Rating vs. Sweetness')
+plt.xlabel('Sweetness')
+plt.ylabel('Rating');
+```
+
+
+![png](bias_cv_files/bias_cv_12_0.png)
+
+
+Using degree 10 polynomial features on 9 random points from the dataset result in a perfectly accurate model for those data points. Unfortunately, this model fails to generalize to previously unseen data from the population.
+
+
+```python
+# HIDDEN
+from sklearn.preprocessing import PolynomialFeatures
+from sklearn.linear_model import LinearRegression
+
+ice2 = pd.read_csv('icecream.csv')
+trans_ten = PolynomialFeatures(degree=10)
+X_ten = trans_ten.fit_transform(ice2[['sweetness']])
+y = ice2['overall']
+
+clf_ten = LinearRegression(fit_intercept=False).fit(X_ten, y)
+```
+
+
+```python
+# HIDDEN
+np.random.seed(1)
+x_devs = np.random.normal(scale=0.4, size=len(ice2))
+y_devs = np.random.normal(scale=0.4, size=len(ice2))
+
+plt.figure(figsize=(10, 5))
+
+plt.subplot(121)
+plt.scatter(ice2['sweetness'], ice2['overall'])
+xs = np.linspace(3.5, 12.5, 1000).reshape(-1, 1)
+ys = clf_ten.predict(trans_ten.transform(xs))
+plt.plot(xs, ys)
+plt.title('Degree 10 polynomial fit')
+plt.ylim(3, 7);
+
+plt.subplot(122)
+ys = clf_ten.predict(trans_ten.transform(xs))
+plt.plot(xs, ys)
+plt.scatter(ice2['sweetness'] + x_devs,
+            ice2['overall'] + y_devs,
+            c='g')
+plt.title('Degree 10 poly, second set of data')
+plt.ylim(3, 7);
+```
+
+
+![png](bias_cv_files/bias_cv_15_0.png)
+
+
+Instead of the above method, we first partition our data into training, validation, and test datasetsusing `scikit-learn`'s [`sklearn.model_selection.train_test_split`](http://scikit-learn.org/stable/modules/generated/sklearn.model_selection.train_test_split.html) method to perform a 70/30% train-test split.
```

Comment:
Space between dataset and using instead of "datasetusing"

Reply:
fixed